### PR TITLE
bakery: return an error if CheckAny is called with no macaroons

### PR DESCRIPTION
**Pull request by rogpeppe**
CheckAny would erroneously succeed when provided with no macaroons.
In most situations this wasn't a problem in practice because most clients
use httpbakery.CheckRequest which does its own check for this condition,
but it's still a potential ravening security hole.

Also add a set of tests for CheckAny because there were none before.

### CODE REVIEW

File: bakery/service.go [hunk 1]
```diff
@@ -144,6 +144,11 @@ func (svc *Service) Check(ms macaroon.Slice, checker FirstPartyChecker) error {
 //
 // It returns any attributes declared in the successfully validated request.
```
Comment (jrwren) [inline bakery/service.go]: Document that it returns an error if mss is empty.

**Comment by mhilton** [comment]
:+1: LGTM

**Comment by jrwren** [comment]
:+1:
